fixes/consolidate-multiple-richtext-history.sql

```diff
@@ -0,0 +1,145 @@
+/*
+ *	Consolidate field history records, on configurable time threshold (@timeThreshold) in minutes.
+ *
+ *	NOTE:  This script defaults to rolling back changes.
+ *		To commit changes, set @saveChanges = 1.
+ */
+
+declare @saveChanges bit = 0,
+@table sysname = 'Test',
+@field varchar(max) = 'ExpectedResults',
+@timeThreshold real = 5,
+@colsAB varchar(max)
+
+select @colsAB=(
+	select REPLACE(' and (A.{col}=B.{col} or (A.{col} is null and B.{col} is null))', '{col}', quotename(COLUMN_NAME))
+	from INFORMATION_SCHEMA.COLUMNS C
+	where C.TABLE_NAME=@table and COLUMN_NAME not in ('ID','AssetType','AuditBegin', @field, 'AuditEnd')
+	for xml path('')
+)
+
+declare @template2 varchar(max) = '
+	declare @error int, @rowcount int
+	set nocount on; begin tran; save tran TX
+
+	create table #assets(ID int not null, CurrentAuditID int not null, NextAuditID int not null)
+
+	-- Asset Rows
+	;with H as (
+		select AA.ID, AA.AssetType, AuditID, R=ROW_NUMBER() OVER(partition by AA.ID, AA.AssetType order by AuditID)
+		from dbo.AssetAudit AA
+	)
+	insert #assets(ID, CurrentAuditID, NextAuditID)
+	select aH.ID, aH.AuditID, bH.AuditID
+	from H as aH
+	join dbo.[@table] aAsset on aAsset.ID=aH.ID and aAsset.AssetType=aH.AssetType and aH.AuditID = aAsset.AuditBegin
+	join H as bH on bH.ID=aH.ID and aH.AssetType=bH.AssetType and bH.R=aH.R+1
+	join dbo.[@table] bAsset on bAsset.ID=bH.ID and bAsset.AssetType=bH.AssetType and bH.AuditID = bAsset.AuditBegin
+	if (@saveChanges != 1) select NULL as ''#assets'', * from #assets
```

Comment:
Quoting aliases with single-ticks is so idiosyncratic that I just learned it works!